# Lab-4: Probability

### By Mohan Bakshi

In [2]:
# Importing packages
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats
from IPython.display import display

In [3]:
# importing the dataset from a csv file using pandas
df = pd.read_csv('insurance.csv')

In [4]:
print(df.shape)
df.head(10)

(1338, 7)


age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520
5   31  female  25.740         0     no  southeast   3756.62160
6   46  female  33.440         1     no  southeast   8240.58960
7   37  female  27.740         3     no  northwest   7281.50560
8   37    male  29.830         2     no  northeast   6406.41070
9   60  female  25.840         0     no  northwest  28923.13692

### An insurance expert said that overall, the insurance charges for men is higher than the insurance charges for women

In [85]:
isurance_charges_men_avg = df[df.sex == 'male']['charges'].mean()
isurance_charges_women_avg = df[df.sex == 'female']['charges'].mean()
isurance_charges_women_std = df[df.sex == 'female']['charges'].std()

z_score_women = (isurance_charges_men_avg - isurance_charges_women_avg) / isurance_charges_women_std

print(f"Insurance charges for Men Average: {isurance_charges_men_avg}")
print(f"Insurance charges for Women Average: {isurance_charges_women_avg}")
print(f"Insurance charges for Women Standard Deviation: {isurance_charges_women_std}")
print(f"z-score: ",z_score_women)

P_less_than_qual_to_z_score = stats.norm.cdf(z_score_women)
P_greater_than_z_score = 1 - P_less_than_qual_to_z_score
print(f'\n\n P(Z > z_score) = {"{:.6f}".format(P_greater_than_z_score)}')

Insurance charges for Men Average: 13956.751177721893
Insurance charges for Women Average: 12569.578843835347
Insurance charges for Women Standard Deviation: 11128.703800915438
z-score:  0.12464814938936905


 P(Z > z_score) = 0.450401


- <b>Based on the p-value of 0.450401, which is greater than the typical significance level of 0.05 , we would not reject the claim that the insurance charges for men is higher than the insurance charges for women.</b>

### An insurance expert said that overall, the insurance charges for people in ‘southwest’ region is higher than the insurance charges for people in the ‘northwest’ region

In [86]:
isurance_charges_southwest_avg = df[df.region == 'southwest']['charges'].mean()
isurance_charges_northwest_avg = df[df.region == 'northwest']['charges'].mean()
isurance_charges_northwest_std = df[df.region == 'northwest']['charges'].std()

z_score_region = (isurance_charges_southwest_avg
                  - isurance_charges_northwest_avg) / isurance_charges_northwest_std

print(f"Insurance charges for Southwest Region Average: {isurance_charges_southwest_avg}")
print(f"Insurance charges for Northwest Region Average: {isurance_charges_northwest_avg}")
print(f"Insurance charges for Northwest Region Standard Deviation: {isurance_charges_northwes_std}")
print(f"z-score: ",z_score_region)


P_less_than_qual_to_z_score = stats.norm.cdf(z_score_region)
P_greater_than_z_score = 1 - P_less_than_qual_to_z_score
print(f'\n\n P(Z > z_score) = {"{:.6f}".format(P_greater_than_z_score)}')

Insurance charges for Southwest Region Average: 12346.937377292308
Insurance charges for Northwest Region Average: 12417.57537396923
Insurance charges for Northwest Region Standard Deviation: 11072.276927579976
z-score:  -0.006379717301052088


 P(Z > z_score) = 0.502545


- Based on the Z-score (-0.006) which indicate that the average insurance charges for the Southwest region are slightly lower than the average charge for the Northwest region. Moreover, Based on Probability interpretation P(Z > z_score), The probability that the insurance charges for the Southwest Region are greater than the charges for the Northwest Region is 0.502545 (approximately 50.25%). This means that there's a 50.25% chance that the average charges for the Southwest Region are greater than those for the Northwest Region.
  
- However, Since the p-value of 0.502545, which is greater than the typical significance level of 0.05 , we could not reject the claim that the insurance charges for people in ‘southwest’ region is higher than the insurance charges for people in the ‘northwest’ region.


### An insurance expert said that overall, the insurance charges for people having 0 children is higher than the insurance charges for people having children.

In [87]:
isurance_charges_zero_children_avg = df[df.children == 0]['charges'].mean()
isurance_charges_non_zero_children_avg = df[df.children != 0]['charges'].mean()
isurance_charges_non_zero_children_std = df[df.children != 0]['charges'].std()

z_score_children = (isurance_charges_zero_children_avg
                  - isurance_charges_non_zero_children_avg) / isurance_charges_non_zero_children_std

print(f"Insurance charges for Zero Children Average: {isurance_charges_zero_children_avg}")
print(f"Insurance charges for Non Zero Children Average: {isurance_charges_non_zero_children_avg}")
print(f"Insurance charges for Non Zero Children Standard Deviation: {isurance_charges_non_zero_children_std}")
print(f"z-score: ",z_score_children)


P_less_than_qual_to_z_score = stats.norm.cdf(z_score_children)
P_greater_than_z_score = 1 - P_less_than_qual_to_z_score
print(f'\n\n P(Z > z_score) = {"{:.6f}".format(P_greater_than_z_score)}')

Insurance charges for Zero Children Average: 12365.97560163589
Insurance charges for Non Zero Children Average: 13949.941093481675
Insurance charges for Non Zero Children Standard Deviation: 12138.305910566816
z-score:  -0.13049312676053818


 P(Z > z_score) = 0.551912


- Based on the Z-score (-0.130) which indicate the average insurance charges for people with Zero Children are slightly lower than the average charges for people with Childrens, but the difference is very small. Moreover, Based on Probability interpretation P(Z > z_score), The probability that the insurance charges for people with Zero Children are greater than the charges for people with Childrens is 0.551912 (approximately 55.2%). This means that there's a 55.2% chance that the average charges for people with Zero Children are greater than those for people with Childrens.
  
- However, Since the p-value is greater than the significance value (0.551912 > 0.05), we could not reject the claim that the insurance charges for people having 0 children is higher than the insurance charges for people having childrens.
